cmd/observe/flows_filter.go

```diff
@@ -546,6 +547,13 @@ func (of *flowFilter) set(f *filterTracker, name, val string, track bool) error
 			f.HttpUrl = append(f.GetHttpUrl(), val)
 		})
 
+	case "http-header":
+		f.apply(func(f *flowpb.FlowFilter) {
+			h := strings.Split(val, ":")
```

Comment:
The header value can contain a colon, so this should be `strings.SplitN(val, ":", 2)` I believe. Would be nice to have a test for such case too.

Reply:
Nice catch! I think `SplitN` won't help here. I would just re-concatenate the result again except of the last item (that will be the value).

Reply:
I changed it like that and added a test-case header to cover it as well. Let me know what you think.

Reply:
Fixed properly after the correction at https://github.com/cilium/hubble/pull/1277#discussion_r1409417738